src/Data/Functor/Linear.hs

```diff
@@ -19,23 +19,14 @@
 module Data.Functor.Linear
   ( -- * Data Functor Hierarchy
     Functor(..)
-  , Applicative(..)
   , (<$>)
   , (<$)
+  , void
```

Comment:
It's a reasonable place for this. No need to re-export it in the prelude, since there can be a `Control.void` as well, and there is little reason to favour one over the other.

In an ideal world, the prelude would re-export the `Data` and `Control` namespace, but Haskell can't do that yet. So be it.